Dockerfile

```diff
@@ -1,23 +1,42 @@
-FROM node:12-alpine as dependency-builder
+# Using BuildKit is highly recommended
+# DOCKER_BUILDKIT=1 docker build .
 
-# add build tools for other architectures
-# subsequent builds should cache this layer
-RUN apk add make g++ python
+# CACHE = 'fresh' or 'cache' if you want a fresh build, or a cached sqlite3 build
+ARG CACHE=fresh
+
+
+FROM node:12-alpine as base
 
 WORKDIR /usr/src/build
 
+# add build tools for other architectures
+RUN apk add --no-cache make g++ python
+
+
+FROM base as sqlite3-builder
+
+ARG SQLITE3_VERSION="latest"
+
+RUN npm install sqlite3@${SQLITE3_VERSION}
```

Comment:
I have a few concerns that this could install different dependencies than in the development environment because it is not using the lock file. This could make debugging a bit harder.

Reply:
Yea, I was concerned about that too. The reason we can't use the lock file is because it changes when any dependency is changed. So copying it in to an image would invalidate the previous cache we had. I tested it with versions different from the `pakcage.json` and it never broke. That doesn't prevent problems in the future though. This was the simplest way I could think to cache the sqlite3 build.